.github/workflows/baremetal-regression-suite.yml

```diff
@@ -0,0 +1,177 @@
+        import sys
+
+        def extractVersion(versionStr):
+            parts = versionStr.split('.')
+            if len(parts) != 3:
+                print('Invalid version: ' + version)
+                sys.exit(1)
+            if parts[2].lower() == 'x':
+                parts[2] = '0'
+
+            major, minor, point = map(int, parts)
+            return [major, minor, point]
+
+        def getTagVersionForCmd(cmd):
+            versionPattern = re.compile(r'.*([0-9]+\.[0-9]+\.[0-9]+).*')
+
+            # Echo git tag command
+            print('Running: ' + cmd)
+
+            # Print current directory
+            print('Current directory: ' + os.getcwd())
+
+            # List files in current directory
+            print('Files in current directory: ' + str(os.listdir()))
+
+            # Get latest release version
+            gitTagProcess = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
+            (output, err) = gitTagProcess.communicate()
+            gitTagProcessStatus = gitTagProcess.wait()
+
+            if gitTagProcessStatus != 0:
+                print('Unable to retrieve latest git tag. With error: ' + str(err))
+                sys.exit(1)
+
+            latestGitTag = str(output)
+
+            versionMatch = versionPattern.match(latestGitTag)
+            if versionMatch:
+                return extractVersion(versionMatch.group(1))
+            else:
+                print('Unable to extract version from git tag: ' + latestGitTag)
+                sys.exit(2)
+
+        def buildVersionString(version):
+            major, minor, point = map(int, version)
+            return f'{major}.{minor}.{point}'
+
+        def getLatestBranchVersion(branchName):
+
+            latestVersion = getTagVersionForCmd("git tag --list 'hpcc4j_*-release' --sort=-v:refname | head -n 1")
+
+            # If we are merging into master we assume it is going into the next minor release
+            if branchName == 'master':
+                return buildVersionString([latestVersion[0], latestVersion[1] + 2, 0])
+            else:
+                # Extract candidate branch major / minor version
+                candidateBranchPattern = re.compile(r'candidate-([0-9]+\.[0-9]+\.([0-9]+|x)).*')
+                branchVersionMatch = candidateBranchPattern.match(branchName)
+                branchVersion = extractVersion(branchVersionMatch.group(1))
+
+                # Get latest release in branch
+                findLatestBranchVer = "git tag --list 'hpcc4j_" + str(branchVersion[0]) + "." + str(branchVersion[1]) + "*-release' --sort=-v:refname | head -n 1"
+                return getTagVersionForCmd(findLatestBranchVer)
+
+        branch_name = os.environ['BRANCH_NAME']
+        latestVersion = getLatestBranchVersion(branch_name)
+        latestVersionURL = 'https://cdn.hpccsystems.com/releases/CE-Candidate-' + latestVersion + '/bin/platform/hpccsystems-platform-community_' + latestVersion + '-1jammy_amd64_withsymbols.deb'
```

Comment:
we'll have to communicate this dependency to Clo, just so they know to let us know before they plan to move away from this cdn, etc.